base/compiler/typeinfer.jl

```diff
@@ -704,63 +664,52 @@ function type_annotate!(interp::AbstractInterpreter, sv::InferenceState, run_opt
     ssavaluetypes = sv.ssavaluetypes
     slotflags = src.slotflags
     nslots = length(slotflags)
-    undefs = fill(false, nslots)
     any_unreachable = false
 
     # this statement traversal does five things:
-    # 1. introduce temporary `TypedSlot`s that are supposed to be replaced with π-nodes later
-    # 2. mark used-undef slots (required by the `slot2reg` conversion)
-    # 3. mark unreached statements for a bulk code deletion (see issue #7836)
-    # 4. widen slot wrappers (`Conditional` and `MustAlias`) and remove `NOT_FOUND` from `ssavaluetypes`
+    # 1. mark used-undef slots (required by the `slot2reg` conversion)
+    # 2. mark unreached statements for a bulk code deletion (see issue #7836)
+    # 3. widen slot wrappers (`Conditional` and `MustAlias`) and remove `NOT_FOUND` from `ssavaluetypes`
```

Comment:
```suggestion
    # 1. widen slot wrappers (`Conditional` and `MustAlias`) and remove `NOT_FOUND` from `ssavaluetypes`
```
since we no longer perform the bulk deletion here and need the used-undef marking.